pkgs/build-support/libredirect/default.nix

```diff
@@ -32,6 +32,14 @@ else stdenv.mkDerivation rec {
   buildPhase = ''
     runHook preBuild
 
+    ENABLE_TEST_SYSTEM=1
+    if [ "$(cat /proc/self/uid_map)" = "         0          0 4294967295" ]; then
+      echo "Boot namespace detected! Disabling tests that use system()."
+      echo "The system() relies on /bin/sh which may not be compatible with nixpkgs's"
+      echo "glibc, so we can't test this without the sandboxing isolation."
+      ENABLE_TEST_SYSTEM=0
+    fi
+
```

Comment:
Is there any way we can detect the *nix* sandbox specifically?  I couldn't find one, so if you don't know of one either then I guess we can't.

Reply:
Yes, there can be false negatives. Maybe running sandboxless Nix build in non-NixOS singularity container could still try to test system(). Suggestions on better detection are welcome.